plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -1049,8 +1049,26 @@ function wc_print_js() {
  */
 function wc_setcookie( $name, $value, $expire = 0, $secure = false, $httponly = false ) {
 	if ( ! headers_sent() ) {
-		setcookie( $name, $value, $expire, COOKIEPATH ? COOKIEPATH : '/', COOKIE_DOMAIN, $secure, apply_filters( 'woocommerce_cookie_httponly', $httponly, $name, $value, $expire, $secure ) );
-	} elseif ( Constants::is_true( 'WP_DEBUG' ) ) {
+		if ( version_compare( PHP_VERSION, '7.3.0', '>=' ) ) {
+			setcookie(
+				$name,
+				$value,
+				apply_filters(
```

Comment:
1. If possible, we'd like to avoid having a hook that is only available to a subset of PHP versions. To that end, it would be great to relocate this, moving it outside of the `version_compare()` conditional so it is available to everyone—and then apply the resulting values using different techniques, according to PHP version.
2. A docblock would be great, and we could use that to highlight that some properties like `samesite` are only available under PHP 7.3+ (or else add a `@see` link to the relevant PHP docs).
3. This would also avoid repeating the same code for the existing `woocommerce_cookie_httponly` hook twice in the same function.

What do you think?